### PR TITLE
Integration of outline and shadow to label

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/3d-tasks/issues/10661
depend: https://github.com/cocos/engine-extensions/pull/345

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/components/label-shadow.ts [hunk 1]
```diff
@@ -77,7 +68,10 @@ export class LabelShadow extends Component {
         }
 
         this._color.set(value);
```
Comment (SantyWang) [inline cocos/2d/components/label-shadow.ts]: No needs to cache color anymore, just set label's property directly

File: cocos/2d/components/label-shadow.ts [hunk 2]
```diff
@@ -100,7 +90,10 @@ export class LabelShadow extends Component {
 
     set offset (value) {
         this._offset = value;
```
Comment (SantyWang) [inline cocos/2d/components/label-shadow.ts]: No needs to cache color anymore, just set label's property directly

File: cocos/2d/components/label-outline.ts [hunk 3]
```diff
@@ -82,7 +70,10 @@ export class LabelOutline extends Component {
         }
 
         this._color.set(value);
```
Comment (SantyWang) [inline cocos/2d/components/label-outline.ts]: No needs to cache color anymore, just set label's property directly

File: cocos/2d/components/label.ts [hunk 4]
```diff
@@ -594,15 +594,150 @@ export class Label extends UIRenderer {
...
+     ** @en
+     ** Outline effect used to change the display, only for system fonts or TTF fonts.
+     **
+     ** @zh
+     ** 描边效果组件,用于字体描边,只能用于系统字体。
+     **/
+    @editable
+    @displayOrder(19)
+    @tooltip('i18n:label.outline_used')
+    get outlineUsed (): boolean {
```
Comment (SantyWang) [inline cocos/2d/components/label.ts]: Take a look at the names in other software

Comment (LinYunMo) [reply cocos/2d/components/label.ts]: UE： Outlined  enableShadow
Godot： outline_size  shadow_outline_size
UGUI：皆为单独组件

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

File: cocos/2d/assembler/label/ttfUtils.ts [hunk 5]
```diff
@@ -62,25 +68,23 @@ export const ttfUtils =  {
...
+            style.outlineWidth = comp.outlineWidth;
         } else {
             style.isOutlined = false;
         }
 
         // shadow// both
-        let shadowComp = LabelShadow && comp.getComponent(LabelShadow);
-        shadowComp = (shadowComp && shadowComp.enabled) ? shadowComp : null;
-        if (shadowComp) {
+        const isShadow = comp.enableShadow && (comp.shadowBlur > 0 || comp.shadowOffset.x !== 0 || comp.shadowOffset.y !== 0);
```
Comment (SantyWang) [inline cocos/2d/assembler/label/ttfUtils.ts]: Use `approx` instead!

File: cocos/2d/components/label-outline.ts [hunk 6]
```diff
@@ -92,37 +78,47 @@ export class LabelOutline extends Component {
...
+        const label = this.node.getComponent(Label);
+        assertIsTrue(label, 'Required Label component to work, please add Label component.');
+        label.enableOutline = false;
     }
 
-    protected _updateRenderData (): void {
+    /**
+     * @deprecated since v3.8.2, please use [[Label.enableOutline]] instead.
+     */
+    public onLoad (): void {
```
Comment (SantyWang) [inline cocos/2d/components/label-outline.ts]: seems not needs onLoad anymore.

File: cocos/2d/components/label.ts [hunk 7]
```diff
@@ -594,15 +594,150 @@ export class Label extends UIRenderer {
...
         this._underlineHeight = value;
         this.markForUpdateRenderData();
     }
 
+    /**
+     ** @en
+     ** Outline effect used to change the display, only for system fonts or TTF fonts.
+     **
+     ** @zh
+     ** 描边效果组件,用于字体描边,只能用于系统字体。
```
Comment (SantyWang) [inline cocos/2d/components/label.ts]: description is not correct. And should hide it when font is BMFont?

**Review by SantyWang — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**